awx/ui/src/components/Search/Search.js

```diff
@@ -65,6 +65,26 @@ function Search({
   const [searchValue, setSearchValue] = useState('');
   const [isFilterDropdownOpen, setIsFilterDropdownOpen] = useState(false);
 
+  const params = parseQueryString(qsConfig, location.search);
+  if (params?.host_filter) {
+    params.ansible_facts = params.host_filter.substring(
+      'ansible_facts__'.length
+    );
+    delete params.host_filter;
+  }
+
+  const searchChips = getChipsByKey(params, columns, qsConfig);
+  const [chipsByKey, setChipsByKey] = useState(
+    JSON.parse(JSON.stringify(searchChips))
+  );
+
+  useEffect(() => {
+    Object.keys(chipsByKey).forEach((el) => {
+      chipsByKey[el].chips = [];
+    });
+    setChipsByKey({ ...chipsByKey, ...searchChips });
+  }, [location.search, searchChips.length]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
```suggestion
  useEffect(() => {
    Object.keys(chipsByKey).forEach((el) => {
      chipsByKey[el].chips = [];
    });
    setChipsByKey(getChipsByKey(params, columns, qsConfig));
  }, [location.search]);
```

searchChips is an object so it looks like searchChips.length is always going to be undefined.  if we can remove that dependency from the array then I think we can also remove the commented `eslint-disable-line...`

Finally, since we want to keep chipsByKey in sync with searchChips lets just use the getChipsByKey function to give us those values here.

Reply:
I agree that `searchChips.length` is unnecessary here and should be removed from the dependency array since it is a js object. 

```
const [chipsByKey, setChipsByKey] = useState(
    JSON.parse(JSON.stringify(searchChips))
);
 ````
Initializing the `chipsByKey` state (above) is key to removing any object reference to `searchChips` and to persisting the original filters. The goal of this PR was to build up the list of filters and maintain them even after the params change. When the user deletes a filter, instead of removing it from the `chipsByKey` object, we pass that specific filter's chips property an empty array. Keeping the original filter key around for PatternFly's Toolbar component was the trick to fixing the bug. 

```
setChipsByKey(getChipsByKey(params, columns, qsConfig));
```
The line above would overwrite the list of filters and bring back the original bug. 

```
setChipsByKey({ ...chipsByKey, ...searchChips });
```
By spreading the original `chipsByKey` and the current `searchChips` into the setter, I'm able to update the state with new search filters from `searchChips` while preserving the older search filter keys.